mlrun/__main__.py

```diff
@@ -373,6 +373,10 @@ def run(
         if auto_mount:
             fn.apply(auto_mount_modifier())
         fn.is_child = from_env and not kfp
+        # if pod is running inside kfp pod, we don't really need the run logs to be printed actively, we can just
+        # pull the run state, and print it at the end
+        # TODO: change watch to be a flag with more options (with_logs, wait_for_completion, etc.)
+        watch = watch or None if kfp else False
```

Comment:
I may be missing something here - if `watch=True` then your entire change will not be activated since you won't use the new `watch=None` option. Only if this was executed with `watch=False` then in kfp we will get the improvement. But, if a step was running within kfp with `watch=False` then it won't wait for logs anyway, won't it?
If I understand correctly the new mode (`watch=None`) then this should be something like:
```
watch = None if kfp and watch
```

Reply:
Well, I figure out it's my bad understanding of the statement which is logically - 
`watch = (watch or None) if kfp else False`
And not (what I thought) -
`watch = watch or (None if kfp else False)`
Maybe it's better to use my suggestion above just to make it clearer...